flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -1219,4 +1231,21 @@ public static void setJobManagerAddressInConfig(Configuration config, InetSocket
 		return constructor.newInstance(params);
 	}
 
+	private static class ClusterShutdownHook extends Thread {
+		private final ClusterClient<?> client;
+
+		ClusterShutdownHook(ClusterClient<?> client) {
```

Comment:
here use `@Nonnull` looks better to me

Reply:
Thanks @yanghua for reviewing. Do you mean @Nonnull in addition to Objects.requireNonNull() or as replacement?

Reply:
Not replace, keep both of them.